### PR TITLE
CRM_Admin_Form - Automatically redirect to the "browse" page after submission

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
This fixes the bug @totten noticed in https://github.com/civicrm/civicrm-core/pull/26884#issuecomment-1643100433 for 2 forms, and more can be fixed just by adding a line of metadata.

Before
------------
- Browse an admin screen that has been refactored for the form to work at a standalone url, e.g. "Location Types" (you can try this with or without the AdminUI extension enabled).
- Right-click "edit" on a location type and open the link in a new tab/window.
- Click "Save"
- Instead of being returned to the browse screen, you go to the CiviCRM homepage.

After
----------
Fixed (for 2 screens so far: Location Types and Scheduled Reminders).

Technical Details
---------
I've patched `CRM_Admin_Form` to read from the paths metadata and setUserContext if a "browse" path exists. Now we just have to add those to every entity...

### CODE REVIEW

**Comment by aydun** [comment]
Following the instructions above, the new tab shows:
```
    Deprecated function: explode(): Passing null to parameter #2 ($string) of type string is deprecated in CRM_Utils_String::extractURLVarValue() (line 385 of /home/jenkins/bknix-dfl/build/core-26887-644k0/web/sites/all/modules/civicrm/CRM/Utils/String.php).
    Deprecated function: explode(): Passing null to parameter #2 ($string) of type string is deprecated in CRM_Utils_String::extractURLVarValue() (line 385 of /home/jenkins/bknix-dfl/build/core-26887-644k0/web/sites/all/modules/civicrm/CRM/Utils/String.php).
``` 

Other than that, it runs as described.

**Comment by aydun** [comment]
Noticed that the warning does not show the first time a new tab is opened, but does on subsequent tab opens.

**Comment by colemanw** [comment]
@aydun ok I pushed up a fix

**Comment by yashodha** [comment]
@colemanw do you want to squash the commits before the merge?

File: CRM/Utils/String.php [hunk 1]
```diff
@@ -362,7 +362,7 @@ public static function match($url1, $url2) {
     $url2Str = parse_url($url2);
 
     if ($url1Str['path'] == $url2Str['path'] &&
-      self::extractURLVarValue(CRM_Utils_Array::value('query', $url1Str)) == self::extractURLVarValue(CRM_Utils_Array::value('query', $url2Str))
+      self::extractURLVarValue($url1Str['query'] ?? '') == self::extractURLVarValue($url2Str['query'] ?? '')
```
Comment (aydun) [inline CRM/Utils/String.php:365]: Code suggest doesn't want to let me suggest this - not sure why.  I know you didn't change this, but naming something `$url1Str` suggests it is a string - which it isn't. So maybe change all `$url1Str` to `$url1Components` and `$url2Str` to `$url2Components` ?

Comment (colemanw) [reply CRM/Utils/String.php]: Ok @aydun I've cleaned up that function and given the variables less-confusing names.

**Pull request merged**